src/BufferingParser.php

```diff
@@ -87,44 +87,49 @@ private function parseBody(string $body, string $boundary = null): Form
             return new Form([]);
         }
 
-        $exp = \explode("\r\n--$boundary\r\n", $body);
+        $exp = \explode("\r\n--$boundary\r\n", $body, $this->fieldCountLimit);
         $exp[0] = \substr($exp[0], \strlen($boundary) + 4);
         $exp[\count($exp) - 1] = \substr(\end($exp), 0, -\strlen($boundary) - 8);
 
         foreach ($exp as $entry) {
-            list($rawHeaders, $text) = \explode("\r\n\r\n", $entry, 2);
-            $headers = [];
-
-            foreach (\explode("\r\n", $rawHeaders) as $header) {
-                $split = \explode(":", $header, 2);
-                if (!isset($split[1])) {
-                    return new Form([]);
-                }
-                $headers[\strtolower($split[0])] = \trim($split[1]);
+            if (($position = \strpos($entry, "\r\n\r\n")) === false) {
+                throw new ParseException("No header/body boundary found");
```

Comment:
This is a minor BC break, though prior a warning was produced if `\r\n\r\n` was not found.

Reply:
If a warning was previously produced, I think it's fine.